Samples/ExtendedControlAndMetadata/EyeGazeAndBackgroundSegmentation/CameraKsPropertyHelper/PropertyInquiry.cpp

```diff
@@ -0,0 +1,182 @@
+        Windows::Foundation::IPropertyValue property = nullptr;
+
+        KSPROPERTY prop(
+            { 0x1CB79112, 0xC0D2, 0x4213, {0x9C, 0xA6, 0xCD, 0x4F, 0xDB, 0x92, 0x79, 0x72} } /* KSPROPERTYSETID_ExtendedCameraControl */,
+            controlId /* KSPROPERTY_CAMERACONTROL_EXTENDED_* */,
+            0x00000001 /* KSPROPERTY_TYPE_GET */);
+        uint8_t* pProp = reinterpret_cast<uint8_t*>(&prop);
+
+        array_view<uint8_t const> serializedProp = array_view<uint8_t const>(pProp, sizeof(KSPROPERTY));
+        auto getResult = controller.GetDevicePropertyByExtendedId(serializedProp, nullptr);
+        if (getResult.Status() == Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::NotSupported)
+        {
+            return property;
+        }
+        if (getResult.Status() != Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::Success)
+        {
+            throw hresult_invalid_argument(L"Could not retrieve device property " + winrt::to_hstring(controlId) + L" with status: " + winrt::to_hstring((int)getResult.Status()));
+        }
+        auto result = getResult.Value();
+        property = result.as<Windows::Foundation::IPropertyValue>();
+
+        return property;
+    }
+
+    /// <summary>
+    /// Return the result of a GET command of a supported extended control onto the specified VideoDeviceController
+    /// </summary>
+    /// <param name="controller"></param>
+    /// <param name="extendedControlKind"></param>
+    /// <returns></returns>
+    winrt::CameraKsPropertyHelper::IExtendedPropertyPayload PropertyInquiry::GetExtendedControl(winrt::Windows::Media::Devices::VideoDeviceController const& controller, winrt::CameraKsPropertyHelper::ExtendedControlKind const& extendedControlKind)
+    {
+        auto property = GetExtendedCameraControlPayload(controller, static_cast<int>(extendedControlKind));
+        if (property == nullptr)
+        {
+            return nullptr;
+        }
+
+        winrt::com_array<uint8_t> container;
+        IExtendedPropertyPayload payload = nullptr;
+
+        switch (extendedControlKind)
+        {
+            case ExtendedControlKind::KSPROPERTY_CAMERACONTROL_EXTENDED_BACKGROUNDSEGMENTATION:
+            {
+                auto container = winrt::com_array<uint8_t>(sizeof(KSCAMERA_EXTENDEDPROP_HEADER));
+                property.GetUInt8Array(container);
+                KSCAMERA_EXTENDEDPROP_HEADER* payloadHeader = reinterpret_cast<KSCAMERA_EXTENDEDPROP_HEADER*>(&container[0]);
+                if ((payloadHeader->Capability & (ULONGLONG)BackgroundSegmentationCapabilityKind::KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_MASK) != 0)
+                {
+                    payload = make<BackgroundSegmentationPropertyPayload>(property);
+                    break;
+                }
+                // else  fallthrough, BackgroundSegmentation control not implementing mask metadata may return a basic property payload per first draft of the DDI spec
```

Comment:
>  // else  fallthrough, BackgroundSegment [](http://example.com/codeflow?start=14&length=41)

might want to change this to unexpected payload

Reply:
no this is expected, it will just return a basic property if the mask mode is not supported

Reply:
won't this throw an error though?  I do not see the fallback and caller is not catching